src/components/ha-network.ts

```diff
@@ -132,16 +128,16 @@ export class HaNetwork extends LitElement {
     });
   }
 
-  private _handleAdapterCheckboxClick(ev: Event) {
-    const checkbox = ev.currentTarget as HaCheckbox;
-    const adapter_name = (checkbox as any).name;
+  private _handleAdapterSwitchClick(ev: Event) {
+    const switch_ = ev.currentTarget as HaSwitch;
+    const adapter_name = (switch_ as any).adapter;
```

Comment:
`HaSwitch` does not have a `name` property like the `HaCheckbox` does, but adapter attribute contains the correct information. Alas, I'm not sure why that's the case, so this might be incorrect.